tests/test_users.py

```diff
@@ -0,0 +1,201 @@
+                   "LinkNameAlreadyInUseException: link name "
+                   "R->FakeRole_0->test_sete_policy%40test.com is already in use"))
+
+        with self.subTest("An error is raised when adding a role that does not exist."):
+            with self.assertRaises(ad.exceptions.BatchWriteException) as ex:
+               user.add_roles(["ghost_role"])
+               self.assertTrue(ex.response['Error']['Message'].endswith("/ Roles / ghost_role\\' does not exist.'"))
+
+        with self.subTest("A user inherits a roles policies when a role is added to a user."):
+            self.assertListEqual(sorted(user.lookup_policies()), sorted([user.statement, role_statement]))
+
+        with self.subTest("A role is removed from user when remove role is called."):
+            user.remove_roles([role_name])
+            self.assertEqual(user.roles, [])
+
+        with self.subTest("A user has multiple roles when multiple roles are added to user."):
+            user.add_roles(role_names)
+            self.assertEqual(user.roles, role_names)
+
+        with self.subTest("A user inherits multiple role policies when the user has multiple roles."):
+            self.assertListEqual(sorted(user.lookup_policies()),
+                                 sorted([user.statement] + role_statements))
+
+        with self.subTest("A user's roles are listed when a listing a users roles."):
+            self.assertListEqual(sorted(user.roles), sorted(role_names))
+
+        with self.subTest("Multiple roles are removed from a user when a multiple roles are specified for removal."):
+            user.remove_roles(role_names)
+            self.assertEqual(user.roles, [])
+
+    def test_group_and_role(self):
+        """
+        A user inherits policies from groups and roles when the user is apart of a group and assigned a role.
+        """
+        name = "test_sete_policy@test.com"
+        user = User(self.directory, name)
+        test_groups = [(f"group_{i}", f"Group policy_{i}") for i in range(5)]
+        groups = [Group.create(self.directory, *i) for i in test_groups]
+        group_names, group_statements = zip(*test_groups)
+        group_names = sorted(group_names)
+        group_statements = sorted(group_statements)
+        test_roles = [(f"role_{i}", f"Role policy_{i}") for i in range(5)]
+        roles = [Role.create(self.directory, *i) for i in test_roles]
+        role_names, role_statements = zip(*test_roles)
+        role_names = sorted(role_names)
+        role_statements = sorted(role_statements)
+
+        user.add_roles(role_names)
+        user.add_groups(group_names)
+
+        self.assertListEqual(sorted(user.roles), ['default_user'] + role_names)
+        self.assertEqual(user.groups, group_names)
+        self.assertSequenceEqual(sorted(user.lookup_policies()), sorted(
+            [user.statement] + group_statements + role_statements)
+                             )
+
+    @unittest.skip("unfinished and low priority")
```

Comment:
`@unittest.skip("unfinished and low priority")`

Would it be better to leave this in a branch rather than merging it in?  If you'd prefer to leave it in, I think it should have a `TODO`.

Reply:
fixed